### ISSUE TITLE
Unable to install in ubuntu gorilla (20.10)

### ISSUE THREAD

**Issue by cubells** [EasyScreenCast/EasyScreenCast#272]
* Where did you download the extension?
 [ x] Gnome shell extension website
 [x ] From Git (Add commit tag)
* Gnome shell version:
$ aptitude show gnome-shell | grep Vers
Versió: **3.38.0-1ubuntu2**

* Operating System and version:
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	**Ubuntu Groovy Gorilla (development branch)**
Release:	20.10
Codename:	groovy

* Display server: Wayland / X11 / Other
**Wayland**

**Comment by agriffis** [comment]
I'm similarly seeing problems on Fedora 33 beta, which is running GNOME 3.38

Looking glass says it can't find `imports.ui.screencast` ... Looks like this import has gone missing somehow

**Comment by somethingGoneWrong** [comment]
@agriffis @cubells  you can try this ugly patch:

- delete line  24 from extension.js, LibRecorder is not used https://github.com/EasyScreenCast/EasyScreenCast/blob/d168b7d09564c146cb800d09fdc85ab97aa100e1/extension.js#L24
- and change the reference for screencast import at line 18 from utilrecorder.js to something like **imports.dbusServices.screencast.screencastService** https://github.com/EasyScreenCast/EasyScreenCast/blob/d168b7d09564c146cb800d09fdc85ab97aa100e1/utilrecorder.js#L18

**Comment by cubells** [comment]
It does not work.

Oct  4 17:01:55 portatil gnome-shell[51116]: JS ERROR: Extension EasyScreenCast@iacopodeenosee.gmail.com: ImportError: No JS module 'dbusServices' found in search path#012@/home/cubells/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/utilrecorder.js:18:21#012@/home/cubells/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/extension.js:36:22#012_callExtensionInit@resource:///org/gnome/shell/ui/extensionSystem.js:425:13#012loadExtension@resource:///org/gnome/shell/ui/extensionSystem.js:346:27#012_loadExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:595:18#012collectFromDatadirs@resource:///org/gnome/shell/misc/fileUtils.js:27:28#012_loadExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:570:19#012_enableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:604:18#012_sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:635:18#012init@resource:///org/gnome/shell/ui/extensionSystem.js:57:14#012_initializeUI@resource:///org/gnome/shell/ui/main.js

I tried with `imports.dbus.screencast.screencastService` and with `imports.service.utils.dbus.screencast.screencastService` unsuccessfully.

Thanks @somethingGoneWrong

**Comment by somethingGoneWrong** [comment]
sorry i have GS 3.36 not 3.38, try to import the screencast interface directly to utilrecorder.js, replace line 18:
```
const { loadInterfaceXML, loadSubInterfaceXML } = imports.misc.fileUtils;
const ScreencastIface = loadInterfaceXML('org.gnome.Shell.Screencast');
```
https://github.com/EasyScreenCast/EasyScreenCast/blob/d168b7d09564c146cb800d09fdc85ab97aa100e1/utilrecorder.js#L18
and change line 29:
```
const ScreenCastProxy = Gio.DBusProxy.makeProxyWrapper(
    ScreencastIface
);
```
https://github.com/EasyScreenCast/EasyScreenCast/blob/d168b7d09564c146cb800d09fdc85ab97aa100e1/utilrecorder.js#L29

**Comment by cubells** [comment]
thanks @somethingGoneWrong 

Now I've got this error:

Oct  5 09:29:51 portatil gnome-shell[26211]: JS ERROR: Extension EasyScreenCast@iacopodeenosee.gmail.com: ImportError: No JS module 'screencast' found in search path#012@/home/cubells/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/extension.js:24:21#012_callExtensionInit@resource:///org/gnome/shell/ui/extensionSystem.js:425:13#012loadExtension@resource:///org/gnome/shell/ui/extensionSystem.js:346:27#012_loadExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:595:18#012collectFromDatadirs@resource:///org/gnome/shell/misc/fileUtils.js:27:28#012_loadExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:570:19#012_enableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:604:18#012_sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:635:18#012init@resource:///org/gnome/shell/ui/extensionSystem.js:57:14#012_initializeUI@resource:///org/gnome/shell/ui/main.js:272:22#012start@resource:///org/gnome/shell/ui/main.js:162:5#012@<main>:1:47

**Comment by somethingGoneWrong** [comment]
you need to delete line 24 on file extension.js like I said [here](https://github.com/EasyScreenCast/EasyScreenCast/issues/272#issuecomment-703263540)
https://github.com/EasyScreenCast/EasyScreenCast/blob/3252312c408e687d1f89342951761e1df8536828/extension.js#L24
and restart gnome shell, If this shouldn't be enough, you'll have to wait for arch to be updated to 3.38 to allow me to do more testing.

**Comment by cubells** [comment]
@somethingGoneWrong 

I've got a different error now:

Oct  5 11:40:09 portatil gnome-shell[8750]: JS ERROR: Extension EasyScreenCast@iacopodeenosee.gmail.com: ImportError: No JS module 'tweener' found in search path#012@/home/cubells/.local/share/gnome-shell/extens
ions/EasyScreenCast@iacopodeenosee.gmail.com/utilnotify.js:17:17#012@/home/cubells/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/selection.js:46:20#012@/home/cubells/.local/share/gn
ome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/utilrecorder.js:25:19#012@/home/cubells/.local/share/gnome-shell/extensions/EasyScreenCast@iacopodeenosee.gmail.com/extension.js:36:22#012_callExtensi
onInit@resource:///org/gnome/shell/ui/extensionSystem.js:425:13#012loadExtension@resource:///org/gnome/shell/ui/extensionSystem.js:346:27#012_loadExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:59
5:18#012collectFromDatadirs@resource:///org/gnome/shell/misc/fileUtils.js:27:28#012_loadExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:570:19#012_enableAllExtensions@resource:///org/gnome/shell/ui/
extensionSystem.js:604:18#012_sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:635:18#012init@resource:///org/gnome/shell/ui/extensionSystem.js:57:14#012_initializeUI@resource:///org/gnome/shell/
ui/main.js:272:22#012start@resource:///org/gnome/shell/ui/main.js:162:5#012@<main>:1:47

**Comment by ilkm** [comment]
我曾经解决过这个问题，在Ubuntu的应用商店记录过，贴上一下图片希望对你有帮助
![image](https://user-images.githubusercontent.com/64707972/95116656-ef2be280-0779-11eb-8ff8-3e9f5b409741.png)

**Comment by somethingGoneWrong** [comment]
@cubells replace line [610](https://github.com/EasyScreenCast/EasyScreenCast/blob/3252312c408e687d1f89342951761e1df8536828/extension.js#L610) of extension.js with this:
```
        this.TimeSlider.x_expand = true;
        this.imSliderDelay.actor.add_child(this.TimeSlider.actor);
```
i have some error when try recording, but i have no time to go deeper yet. sry

**Comment by epozuelo** [comment]
I have added a few more fixes in https://github.com/epozuelo/EasyScreenCast/commit/dd675d310cffd199df1db3d625c3f615cb7693cf

It is able to start with GNOME 3.38 and start recording, though it gives an error that I haven't managed to fix yet

**Comment by ilkm** [comment]
Thank you for your contribution




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Emilio Pozuelo Monfort"<notifications@github.com&gt;; 
发送时间: 2020年10月16日(星期五) 凌晨1:44
收件人: "EasyScreenCast/EasyScreenCast"<EasyScreenCast@noreply.github.com&gt;; 
抄送: "极致"<irez@foxmail.com&gt;; "Comment"<comment@noreply.github.com&gt;; 
主题: Re: [EasyScreenCast/EasyScreenCast] Unable to install in ubuntu gorilla (20.10) (#272)





 
I have added a few more fixes in epozuelo@dd675d3
 
It is able to start with GNOME 3.38 and start recording, though it gives an error that I haven't managed to fix yet

**Comment by adangel** [comment]
I think, this is fixed with 1.3.0
